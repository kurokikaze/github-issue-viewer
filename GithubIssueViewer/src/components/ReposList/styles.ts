import {StyleSheet} from 'react-native';

const styles = StyleSheet.create({
  baseContainer: {
    display: 'flex',
    flex: 1,
    flexGrow: 1,
    flexDirection: 'column',
    padding: 40,
  },
  reposContainer: {
    flex: 1,
    display: 'flex',
    flexDirection: 'column',
  },
  repoName: {
    fontSize: 18,
    lineHeight: 20,
    fontWeight: 'normal',
    display: 'flex',
    flex: 1,
    flexWrap: 'wrap',
    marginTop: 18,
  },
  overlayIndicator: {
    position: 'absolute',
    left: 0,
    right: 0,
    top: 0,
    bottom: 0,
    alignItems: 'center',
    justifyContent: 'center',
  },
});

export default styles;
