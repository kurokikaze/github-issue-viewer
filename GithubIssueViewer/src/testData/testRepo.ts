import {GithubRepositoryResponse, GithubUserResponse} from '../types';

const testOwner: GithubUserResponse = {
  login: 'kurokikaze',
  id: 92214,
  node_id: 'MDQ6VXNlcjkyMjE0',
  avatar_url: 'https://avatars.githubusercontent.com/u/92214?v=4',
  gravatar_id: '',
  url: 'https://api.github.com/users/kurokikaze',
  html_url: 'https://github.com/kurokikaze',
  followers_url: 'https://api.github.com/users/kurokikaze/followers',
  following_url:
    'https://api.github.com/users/kurokikaze/following{/other_user}',
  gists_url: 'https://api.github.com/users/kurokikaze/gists{/gist_id}',
  starred_url: 'https://api.github.com/users/kurokikaze/starred{/owner}{/repo}',
  subscriptions_url: 'https://api.github.com/users/kurokikaze/subscriptions',
  organizations_url: 'https://api.github.com/users/kurokikaze/orgs',
  repos_url: 'https://api.github.com/users/kurokikaze/repos',
  events_url: 'https://api.github.com/users/kurokikaze/events{/privacy}',
  received_events_url:
    'https://api.github.com/users/kurokikaze/received_events',
  type: 'User',
  site_admin: false,
};

export const testRepo: GithubRepositoryResponse = {
  id: 215071064,
  node_id: 'MDEwOlJlcG9zaXRvcnkyMTUwNzEwNjQ=',
  name: 'amadeus-bot',
  full_name: 'kurokikaze/amadeus-bot',
  private: false,
  owner: testOwner,
  html_url: 'https://github.com/kurokikaze/amadeus-bot',
  description: 'Bot for "Unleash the Geek" challenge',
  fork: false,
  url: 'https://api.github.com/repos/kurokikaze/amadeus-bot',
  forks_url: 'https://api.github.com/repos/kurokikaze/amadeus-bot/forks',
  keys_url: 'https://api.github.com/repos/kurokikaze/amadeus-bot/keys{/key_id}',
  collaborators_url:
    'https://api.github.com/repos/kurokikaze/amadeus-bot/collaborators{/collaborator}',
  teams_url: 'https://api.github.com/repos/kurokikaze/amadeus-bot/teams',
  hooks_url: 'https://api.github.com/repos/kurokikaze/amadeus-bot/hooks',
  issue_events_url:
    'https://api.github.com/repos/kurokikaze/amadeus-bot/issues/events{/number}',
  events_url: 'https://api.github.com/repos/kurokikaze/amadeus-bot/events',
  assignees_url:
    'https://api.github.com/repos/kurokikaze/amadeus-bot/assignees{/user}',
  branches_url:
    'https://api.github.com/repos/kurokikaze/amadeus-bot/branches{/branch}',
  tags_url: 'https://api.github.com/repos/kurokikaze/amadeus-bot/tags',
  blobs_url:
    'https://api.github.com/repos/kurokikaze/amadeus-bot/git/blobs{/sha}',
  git_tags_url:
    'https://api.github.com/repos/kurokikaze/amadeus-bot/git/tags{/sha}',
  git_refs_url:
    'https://api.github.com/repos/kurokikaze/amadeus-bot/git/refs{/sha}',
  trees_url:
    'https://api.github.com/repos/kurokikaze/amadeus-bot/git/trees{/sha}',
  statuses_url:
    'https://api.github.com/repos/kurokikaze/amadeus-bot/statuses/{sha}',
  languages_url:
    'https://api.github.com/repos/kurokikaze/amadeus-bot/languages',
  stargazers_url:
    'https://api.github.com/repos/kurokikaze/amadeus-bot/stargazers',
  contributors_url:
    'https://api.github.com/repos/kurokikaze/amadeus-bot/contributors',
  subscribers_url:
    'https://api.github.com/repos/kurokikaze/amadeus-bot/subscribers',
  subscription_url:
    'https://api.github.com/repos/kurokikaze/amadeus-bot/subscription',
  commits_url:
    'https://api.github.com/repos/kurokikaze/amadeus-bot/commits{/sha}',
  git_commits_url:
    'https://api.github.com/repos/kurokikaze/amadeus-bot/git/commits{/sha}',
  comments_url:
    'https://api.github.com/repos/kurokikaze/amadeus-bot/comments{/number}',
  issue_comment_url:
    'https://api.github.com/repos/kurokikaze/amadeus-bot/issues/comments{/number}',
  contents_url:
    'https://api.github.com/repos/kurokikaze/amadeus-bot/contents/{+path}',
  compare_url:
    'https://api.github.com/repos/kurokikaze/amadeus-bot/compare/{base}...{head}',
  merges_url: 'https://api.github.com/repos/kurokikaze/amadeus-bot/merges',
  archive_url:
    'https://api.github.com/repos/kurokikaze/amadeus-bot/{archive_format}{/ref}',
  downloads_url:
    'https://api.github.com/repos/kurokikaze/amadeus-bot/downloads',
  issues_url:
    'https://api.github.com/repos/kurokikaze/amadeus-bot/issues{/number}',
  pulls_url:
    'https://api.github.com/repos/kurokikaze/amadeus-bot/pulls{/number}',
  milestones_url:
    'https://api.github.com/repos/kurokikaze/amadeus-bot/milestones{/number}',
  notifications_url:
    'https://api.github.com/repos/kurokikaze/amadeus-bot/notifications{?since,all,participating}',
  labels_url:
    'https://api.github.com/repos/kurokikaze/amadeus-bot/labels{/name}',
  releases_url:
    'https://api.github.com/repos/kurokikaze/amadeus-bot/releases{/id}',
  deployments_url:
    'https://api.github.com/repos/kurokikaze/amadeus-bot/deployments',
  created_at: '2019-10-14T14:52:50Z',
  updated_at: '2019-10-16T08:45:32Z',
  pushed_at: '2019-10-16T08:45:30Z',
  git_url: 'git://github.com/kurokikaze/amadeus-bot.git',
  ssh_url: 'git@github.com:kurokikaze/amadeus-bot.git',
  clone_url: 'https://github.com/kurokikaze/amadeus-bot.git',
  svn_url: 'https://github.com/kurokikaze/amadeus-bot',
  homepage: null,
  size: 14,
  stargazers_count: 0,
  watchers_count: 0,
  language: 'JavaScript',
  has_issues: true,
  has_projects: true,
  has_downloads: true,
  has_wiki: true,
  has_pages: false,
  forks_count: 0,
  mirror_url: null,
  archived: false,
  disabled: false,
  open_issues_count: 0,
  license: null,
  forks: 0,
  open_issues: 0,
  watchers: 0,
  default_branch: 'master',
};

export const anotherTestRepo: GithubRepositoryResponse = {
  id: 155540694,
  node_id: 'MDEwOlJlcG9zaXRvcnkxNTU1NDA2OTQ=',
  name: 'andor-vis',
  full_name: 'kurokikaze/andor-vis',
  private: false,
  owner: testOwner,
  html_url: 'https://github.com/kurokikaze/andor-vis',
  description: 'Andor map graph and visualisation',
  fork: false,
  url: 'https://api.github.com/repos/kurokikaze/andor-vis',
  forks_url: 'https://api.github.com/repos/kurokikaze/andor-vis/forks',
  keys_url: 'https://api.github.com/repos/kurokikaze/andor-vis/keys{/key_id}',
  collaborators_url:
    'https://api.github.com/repos/kurokikaze/andor-vis/collaborators{/collaborator}',
  teams_url: 'https://api.github.com/repos/kurokikaze/andor-vis/teams',
  hooks_url: 'https://api.github.com/repos/kurokikaze/andor-vis/hooks',
  issue_events_url:
    'https://api.github.com/repos/kurokikaze/andor-vis/issues/events{/number}',
  events_url: 'https://api.github.com/repos/kurokikaze/andor-vis/events',
  assignees_url:
    'https://api.github.com/repos/kurokikaze/andor-vis/assignees{/user}',
  branches_url:
    'https://api.github.com/repos/kurokikaze/andor-vis/branches{/branch}',
  tags_url: 'https://api.github.com/repos/kurokikaze/andor-vis/tags',
  blobs_url:
    'https://api.github.com/repos/kurokikaze/andor-vis/git/blobs{/sha}',
  git_tags_url:
    'https://api.github.com/repos/kurokikaze/andor-vis/git/tags{/sha}',
  git_refs_url:
    'https://api.github.com/repos/kurokikaze/andor-vis/git/refs{/sha}',
  trees_url:
    'https://api.github.com/repos/kurokikaze/andor-vis/git/trees{/sha}',
  statuses_url:
    'https://api.github.com/repos/kurokikaze/andor-vis/statuses/{sha}',
  languages_url: 'https://api.github.com/repos/kurokikaze/andor-vis/languages',
  stargazers_url:
    'https://api.github.com/repos/kurokikaze/andor-vis/stargazers',
  contributors_url:
    'https://api.github.com/repos/kurokikaze/andor-vis/contributors',
  subscribers_url:
    'https://api.github.com/repos/kurokikaze/andor-vis/subscribers',
  subscription_url:
    'https://api.github.com/repos/kurokikaze/andor-vis/subscription',
  commits_url:
    'https://api.github.com/repos/kurokikaze/andor-vis/commits{/sha}',
  git_commits_url:
    'https://api.github.com/repos/kurokikaze/andor-vis/git/commits{/sha}',
  comments_url:
    'https://api.github.com/repos/kurokikaze/andor-vis/comments{/number}',
  issue_comment_url:
    'https://api.github.com/repos/kurokikaze/andor-vis/issues/comments{/number}',
  contents_url:
    'https://api.github.com/repos/kurokikaze/andor-vis/contents/{+path}',
  compare_url:
    'https://api.github.com/repos/kurokikaze/andor-vis/compare/{base}...{head}',
  merges_url: 'https://api.github.com/repos/kurokikaze/andor-vis/merges',
  archive_url:
    'https://api.github.com/repos/kurokikaze/andor-vis/{archive_format}{/ref}',
  downloads_url: 'https://api.github.com/repos/kurokikaze/andor-vis/downloads',
  issues_url:
    'https://api.github.com/repos/kurokikaze/andor-vis/issues{/number}',
  pulls_url: 'https://api.github.com/repos/kurokikaze/andor-vis/pulls{/number}',
  milestones_url:
    'https://api.github.com/repos/kurokikaze/andor-vis/milestones{/number}',
  notifications_url:
    'https://api.github.com/repos/kurokikaze/andor-vis/notifications{?since,all,participating}',
  labels_url: 'https://api.github.com/repos/kurokikaze/andor-vis/labels{/name}',
  releases_url:
    'https://api.github.com/repos/kurokikaze/andor-vis/releases{/id}',
  deployments_url:
    'https://api.github.com/repos/kurokikaze/andor-vis/deployments',
  created_at: '2018-10-31T10:42:36Z',
  updated_at: '2018-10-31T12:18:29Z',
  pushed_at: '2018-10-31T12:18:28Z',
  git_url: 'git://github.com/kurokikaze/andor-vis.git',
  ssh_url: 'git@github.com:kurokikaze/andor-vis.git',
  clone_url: 'https://github.com/kurokikaze/andor-vis.git',
  svn_url: 'https://github.com/kurokikaze/andor-vis',
  homepage: null,
  size: 2,
  stargazers_count: 0,
  watchers_count: 0,
  language: 'JavaScript',
  has_issues: true,
  has_projects: true,
  has_downloads: true,
  has_wiki: true,
  has_pages: false,
  forks_count: 0,
  mirror_url: null,
  archived: false,
  disabled: false,
  open_issues_count: 5,
  license: null,
  forks: 0,
  open_issues: 0,
  watchers: 0,
  default_branch: 'master',
};
