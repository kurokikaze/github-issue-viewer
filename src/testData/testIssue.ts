import {GithubIssueResponse} from '../types';

export const testIssue: GithubIssueResponse = {
  url: 'https://api.github.com/repos/github/backup-utils/issues/770',
  repository_url: 'https://api.github.com/repos/github/backup-utils',
  labels_url:
    'https://api.github.com/repos/github/backup-utils/issues/770/labels{/name}',
  comments_url:
    'https://api.github.com/repos/github/backup-utils/issues/770/comments',
  events_url:
    'https://api.github.com/repos/github/backup-utils/issues/770/events',
  html_url: 'https://github.com/github/backup-utils/pull/770',
  id: 987963032,
  node_id: 'MDExOlB1bGxSZXF1ZXN0NzI2OTcxMTc5',
  number: 770,
  title: 'Clear GitHub Connect settings when not restoring settings',
  user: {
    login: 'lildude',
    id: 627280,
    node_id: 'MDQ6VXNlcjYyNzI4MA==',
    avatar_url: 'https://avatars.githubusercontent.com/u/627280?v=4',
    gravatar_id: '',
    url: 'https://api.github.com/users/lildude',
    html_url: 'https://github.com/lildude',
    followers_url: 'https://api.github.com/users/lildude/followers',
    following_url:
      'https://api.github.com/users/lildude/following{/other_user}',
    gists_url: 'https://api.github.com/users/lildude/gists{/gist_id}',
    starred_url: 'https://api.github.com/users/lildude/starred{/owner}{/repo}',
    subscriptions_url: 'https://api.github.com/users/lildude/subscriptions',
    organizations_url: 'https://api.github.com/users/lildude/orgs',
    repos_url: 'https://api.github.com/users/lildude/repos',
    events_url: 'https://api.github.com/users/lildude/events{/privacy}',
    received_events_url: 'https://api.github.com/users/lildude/received_events',
    type: 'User',
    site_admin: true,
  },
  labels: [],
  state: 'open',
  locked: false,
  assignee: null,
  assignees: [],
  milestone: null,
  comments: 0,
  created_at: '2021-09-03T17:12:00Z',
  updated_at: '2021-09-08T16:27:33Z',
  closed_at: null,
  author_association: 'MEMBER',
  active_lock_reason: null,
  pull_request: {
    url: 'https://api.github.com/repos/github/backup-utils/pulls/770',
    html_url: 'https://github.com/github/backup-utils/pull/770',
    diff_url: 'https://github.com/github/backup-utils/pull/770.diff',
    patch_url: 'https://github.com/github/backup-utils/pull/770.patch',
  },
  body: "At the moment, when a backup it restored _without_ any settings, ie without using the `--config` flag, the GitHub Connect settings are still restored. This is because they're stored in the MySQL database and there's nothing in place to clear them or prevent them from being restored.\r\n\r\nThis leads to a problem where the restored instance will have all the same settings and GitHub Connect-specific unique ID as the original host and will start competing with the original instance if both are active. This can result in one instance being kicked off GitHub Connect, or confusing license sync information, if both remain connected.\r\n\r\nThis PR resolves this problem by calling a new script immediately after the MySQL database has been restored. This script will be added to all currently supported versions of GHES.",
  performed_via_github_app: null,
};
