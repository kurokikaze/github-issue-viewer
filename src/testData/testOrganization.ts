import {GithubOrganizationResponse} from '../types';

export const testOrganization: GithubOrganizationResponse = {
  login: 'revelation',
  id: 728,
  node_id: 'MDEyOk9yZ2FuaXphdGlvbjcyOA==',
  url: 'https://api.github.com/orgs/revelation',
  repos_url: 'https://api.github.com/orgs/revelation/repos',
  events_url: 'https://api.github.com/orgs/revelation/events',
  hooks_url: 'https://api.github.com/orgs/revelation/hooks',
  issues_url: 'https://api.github.com/orgs/revelation/issues',
  members_url: 'https://api.github.com/orgs/revelation/members{/member}',
  public_members_url:
    'https://api.github.com/orgs/revelation/public_members{/member}',
  avatar_url: 'https://avatars.githubusercontent.com/u/728?v=4',
  description: '',
};

export const anotherTestOrganization: GithubOrganizationResponse = {
  login: 'cdr',
  id: 1375,
  node_id: 'MDEyOk9yZ2FuaXphdGlvbjEzNzU=',
  url: 'https://api.github.com/orgs/cdr',
  repos_url: 'https://api.github.com/orgs/cdr/repos',
  events_url: 'https://api.github.com/orgs/cdr/events',
  hooks_url: 'https://api.github.com/orgs/cdr/hooks',
  issues_url: 'https://api.github.com/orgs/cdr/issues',
  members_url: 'https://api.github.com/orgs/cdr/members{/member}',
  public_members_url: 'https://api.github.com/orgs/cdr/public_members{/member}',
  avatar_url: 'https://avatars.githubusercontent.com/u/1375?v=4',
  description: 'Dedicated to keeping developers in flow.',
};
