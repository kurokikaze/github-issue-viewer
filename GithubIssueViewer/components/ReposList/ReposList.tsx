import React from 'react';
import {
  ActivityIndicator,
  ScrollView,
  Text,
  useColorScheme,
  View,
} from 'react-native';
import {useSelector} from 'react-redux';
import {getRepos, getLoading} from '../../selectors';
import {GithubRepositoryResponse} from '../../types';
import Section from '../Section/Section';

import styles from './styles';

import {Colors} from 'react-native/Libraries/NewAppScreen';

type ReposListProps = {
  onSelectRepo: (repo: GithubRepositoryResponse) => void;
};

const ACTIVE_REPO_COLOR = '#056aff';

const ReposList = ({onSelectRepo}: ReposListProps) => {
  const githubRepos = useSelector(getRepos);
  const reposLoading = useSelector(getLoading);

  const isDarkMode = useColorScheme() === 'dark';
  const modeColor = isDarkMode ? Colors.white : Colors.black;

  return (
    <Section title={`${githubRepos.length} Repos`}>
      <ScrollView style={styles.reposContainer}>
        {reposLoading && <ActivityIndicator size="large" />}
        {githubRepos.length > 0 &&
          githubRepos.map(repo => (
            <View key={repo.id}>
              <Text
                onPress={() => repo.open_issues_count > 0 && onSelectRepo(repo)}
                style={[
                  styles.repoName,
                  {
                    color:
                      repo.open_issues_count > 0
                        ? ACTIVE_REPO_COLOR
                        : modeColor,
                  },
                ]}>
                {repo.name}{' '}
                {repo.open_issues_count > 0 &&
                  `(${repo.open_issues_count} open  issues)`}
              </Text>
            </View>
          ))}
      </ScrollView>
    </Section>
  );
};

export default ReposList;
