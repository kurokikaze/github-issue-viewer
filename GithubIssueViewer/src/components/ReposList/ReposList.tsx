import React, {useContext} from 'react';
import {
  ActivityIndicator,
  ScrollView,
  Text,
  TextStyle,
  View,
} from 'react-native';
import {useDispatch, useSelector} from 'react-redux';
import {fetchReposPage} from '../../actions';
import {
  getRepos,
  getReposLoading,
  getReposPagination,
  getReposPage,
} from '../../selectors';
import {GithubRepositoryResponse} from '../../types';
import {COLOR_LINKS} from '../../styles';
import {ThemeContext} from '../ThemeContext/ThemeContext';
import {pluralize, isPaginationUsable} from '../../utils';

import styles from './styles';
import {Pagination} from '../Pagination/Pagination';

type ReposListProps = {
  onSelectRepo: (repo: GithubRepositoryResponse) => void;
};

const linkRepoStyle: TextStyle = {
  color: COLOR_LINKS,
};

const ReposList = ({onSelectRepo}: ReposListProps) => {
  const githubRepos = useSelector(getRepos);
  const isLoading = useSelector(getReposLoading);
  const pagination = useSelector(getReposPagination);
  const page = useSelector(getReposPage);

  const dispatch = useDispatch();

  const theme = useContext(ThemeContext);

  return (
    <View style={[styles.baseContainer]}>
      {isPaginationUsable(pagination) ? (
        <Pagination
          links={pagination}
          page={page}
          loading={isLoading}
          onPageChange={newPage => dispatch(fetchReposPage(newPage))}
        />
      ) : null}
      <View>
        <ScrollView style={styles.reposContainer}>
          {githubRepos.length > 0 &&
            githubRepos.map(repo => (
              <View key={repo.id} data-testID={`repo-${repo.id}`}>
                <Text
                  onPress={() => repo.has_issues && onSelectRepo(repo)}
                  data-testID="select-repo"
                  style={[
                    theme.textStyle,
                    styles.repoName,
                    repo.has_issues && linkRepoStyle,
                  ]}>
                  {repo.name}{' '}
                  {repo.open_issues_count > 0 &&
                    `(${pluralize({
                      singular: 'open issue',
                      plural: 'open issues',
                      empty: '',
                      count: repo.open_issues_count,
                    })})`}
                </Text>
              </View>
            ))}
        </ScrollView>
        {isLoading && (
          <ActivityIndicator size={300} style={styles.overlayIndicator} />
        )}
      </View>
      {isPaginationUsable(pagination) ? (
        <Pagination
          links={pagination}
          page={page}
          loading={isLoading}
          onPageChange={newPage =>
            !isLoading && dispatch(fetchReposPage(newPage))
          }
        />
      ) : null}
    </View>
  );
};

export default ReposList;
