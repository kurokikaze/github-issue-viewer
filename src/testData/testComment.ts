import {testOwner} from './testRepo';

export const testComment = {
  url: 'https://api.github.com/repos/github/backup-utils/issues/comments/899664665',
  html_url:
    'https://github.com/github/backup-utils/issues/761#issuecomment-899664665',
  issue_url: 'https://api.github.com/repos/github/backup-utils/issues/761',
  id: 899664665,
  node_id: 'IC_kwDOAUntRM41n8sZ',
  user: testOwner,
  created_at: '2021-08-16T16:56:14Z',
  updated_at: '2021-08-16T16:56:14Z',
  author_association: 'CONTRIBUTOR',
  body: 'Comment body',
  performed_via_github_app: null,
};
