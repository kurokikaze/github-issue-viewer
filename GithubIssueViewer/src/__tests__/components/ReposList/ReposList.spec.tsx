import React from 'react';
import 'react-native';
import {Provider} from 'react-redux';
// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';
import configureMockStore from 'redux-mock-store';

import ReposList from '../../../components/ReposList/ReposList';
import {testRepo, anotherTestRepo} from '../../../testData/testRepo';

const mockStore = configureMockStore([]);

describe('ReposList', () => {
  it('renders correctly', () => {
    const store = mockStore({
      repos: {
        list: [testRepo, anotherTestRepo],
        pagination: {
          first: null,
          prev: null,
          next: null,
          last: null,
        },
        loading: false,
      },
    });

    const wrapper = renderer.create(
      <Provider store={store}>
        <ReposList onSelectRepo={() => null} />
      </Provider>,
    );

    expect(wrapper).toMatchSnapshot();
  });

  it('renders correctly (loading)', () => {
    const store = mockStore({
      repos: {
        list: [testRepo, anotherTestRepo],
        pagination: {
          first: null,
          prev: null,
          next: null,
          last: null,
        },
        loading: true,
      },
    });

    const wrapper = renderer.create(
      <Provider store={store}>
        <ReposList onSelectRepo={() => null} />
      </Provider>,
    );

    expect(wrapper).toMatchSnapshot();
  });

  it('renders correctly (pagination)', () => {
    const store = mockStore({
      repos: {
        list: [testRepo, anotherTestRepo],
        pagination: {
          first: null,
          prev: null,
          next: 2,
          last: 4,
        },
        loading: true,
      },
    });

    const wrapper = renderer.create(
      <Provider store={store}>
        <ReposList onSelectRepo={() => null} />
      </Provider>,
    );

    expect(wrapper).toMatchSnapshot();
  });

  it('onSelectRepo', () => {
    const repoSelectHandler = jest.fn();

    const store = mockStore({
      repos: {
        list: [testRepo, anotherTestRepo],
        pagination: {
          first: null,
          prev: null,
          next: 2,
          last: 4,
        },
        loading: true,
      },
    });

    const wrapper = renderer.create(
      <Provider store={store}>
        <ReposList onSelectRepo={repoSelectHandler} />
      </Provider>,
    );

    wrapper.root
      .findByProps({'data-testID': `repo-${testRepo.id}`})
      .findByProps({'data-testID': 'select-repo'})
      .props.onPress();

    expect(repoSelectHandler).toHaveBeenCalledWith(testRepo);
  });
});
