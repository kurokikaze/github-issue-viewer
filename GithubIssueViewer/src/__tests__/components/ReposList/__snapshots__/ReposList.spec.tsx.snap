// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReposList renders correctly (loading) 1`] = `
<View
  style={
    Array [
      Object {
        "display": "flex",
        "flex": 1,
        "flexDirection": "column",
        "flexGrow": 1,
        "padding": 40,
      },
    ]
  }
>
  <View>
    <RCTScrollView
      style={
        Object {
          "display": "flex",
          "flex": 1,
          "flexDirection": "column",
        }
      }
    >
      <View>
        <View
          data-testID="repo-215071064"
        >
          <Text
            data-testID="select-repo"
            onPress={[Function]}
            style={
              Array [
                Object {
                  "color": "#333",
                },
                Object {
                  "display": "flex",
                  "flex": 1,
                  "flexWrap": "wrap",
                  "fontSize": 18,
                  "fontWeight": "normal",
                  "lineHeight": 20,
                  "marginTop": 18,
                },
                Object {
                  "color": "#056aff",
                },
              ]
            }
          >
            amadeus-bot
             
          </Text>
        </View>
        <View
          data-testID="repo-155540694"
        >
          <Text
            data-testID="select-repo"
            onPress={[Function]}
            style={
              Array [
                Object {
                  "color": "#333",
                },
                Object {
                  "display": "flex",
                  "flex": 1,
                  "flexWrap": "wrap",
                  "fontSize": 18,
                  "fontWeight": "normal",
                  "lineHeight": 20,
                  "marginTop": 18,
                },
                Object {
                  "color": "#056aff",
                },
              ]
            }
          >
            andor-vis
             
            (5 open issues)
          </Text>
        </View>
      </View>
    </RCTScrollView>
    <ActivityIndicator
      size={300}
      style={
        Object {
          "alignItems": "center",
          "bottom": 0,
          "justifyContent": "center",
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    />
  </View>
</View>
`;

exports[`ReposList renders correctly (pagination) 1`] = `
<View
  style={
    Array [
      Object {
        "display": "flex",
        "flex": 1,
        "flexDirection": "column",
        "flexGrow": 1,
        "padding": 40,
      },
    ]
  }
>
  <View
    style={
      Object {
        "display": "flex",
        "flexDirection": "row",
        "justifyContent": "space-between",
      }
    }
  >
    <View>
      <Text
        onPress={[Function]}
        style={
          Array [
            Object {
              "color": "#333",
            },
            Object {
              "fontSize": 36,
            },
            Object {
              "opacity": 0.3,
            },
          ]
        }
      >
        &lt;&lt;
      </Text>
    </View>
    <View>
      <Text
        onPress={[Function]}
        style={
          Array [
            Object {
              "color": "#333",
            },
            Object {
              "fontSize": 36,
            },
            Object {
              "opacity": 0.3,
            },
          ]
        }
      >
        &lt;
      </Text>
    </View>
    <View>
      <ActivityIndicator
        size={36}
      />
    </View>
    <View>
      <Text
        onPress={[Function]}
        style={
          Array [
            Object {
              "color": "#333",
            },
            Object {
              "fontSize": 36,
            },
            false,
          ]
        }
      >
        &gt;
      </Text>
    </View>
    <View>
      <Text
        onPress={[Function]}
        style={
          Array [
            Object {
              "color": "#333",
            },
            Object {
              "fontSize": 36,
            },
            false,
          ]
        }
      >
        &gt;&gt;
      </Text>
    </View>
  </View>
  <View>
    <RCTScrollView
      style={
        Object {
          "display": "flex",
          "flex": 1,
          "flexDirection": "column",
        }
      }
    >
      <View>
        <View
          data-testID="repo-215071064"
        >
          <Text
            data-testID="select-repo"
            onPress={[Function]}
            style={
              Array [
                Object {
                  "color": "#333",
                },
                Object {
                  "display": "flex",
                  "flex": 1,
                  "flexWrap": "wrap",
                  "fontSize": 18,
                  "fontWeight": "normal",
                  "lineHeight": 20,
                  "marginTop": 18,
                },
                Object {
                  "color": "#056aff",
                },
              ]
            }
          >
            amadeus-bot
             
          </Text>
        </View>
        <View
          data-testID="repo-155540694"
        >
          <Text
            data-testID="select-repo"
            onPress={[Function]}
            style={
              Array [
                Object {
                  "color": "#333",
                },
                Object {
                  "display": "flex",
                  "flex": 1,
                  "flexWrap": "wrap",
                  "fontSize": 18,
                  "fontWeight": "normal",
                  "lineHeight": 20,
                  "marginTop": 18,
                },
                Object {
                  "color": "#056aff",
                },
              ]
            }
          >
            andor-vis
             
            (5 open issues)
          </Text>
        </View>
      </View>
    </RCTScrollView>
    <ActivityIndicator
      size={300}
      style={
        Object {
          "alignItems": "center",
          "bottom": 0,
          "justifyContent": "center",
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "display": "flex",
        "flexDirection": "row",
        "justifyContent": "space-between",
      }
    }
  >
    <View>
      <Text
        onPress={[Function]}
        style={
          Array [
            Object {
              "color": "#333",
            },
            Object {
              "fontSize": 36,
            },
            Object {
              "opacity": 0.3,
            },
          ]
        }
      >
        &lt;&lt;
      </Text>
    </View>
    <View>
      <Text
        onPress={[Function]}
        style={
          Array [
            Object {
              "color": "#333",
            },
            Object {
              "fontSize": 36,
            },
            Object {
              "opacity": 0.3,
            },
          ]
        }
      >
        &lt;
      </Text>
    </View>
    <View>
      <ActivityIndicator
        size={36}
      />
    </View>
    <View>
      <Text
        onPress={[Function]}
        style={
          Array [
            Object {
              "color": "#333",
            },
            Object {
              "fontSize": 36,
            },
            false,
          ]
        }
      >
        &gt;
      </Text>
    </View>
    <View>
      <Text
        onPress={[Function]}
        style={
          Array [
            Object {
              "color": "#333",
            },
            Object {
              "fontSize": 36,
            },
            false,
          ]
        }
      >
        &gt;&gt;
      </Text>
    </View>
  </View>
</View>
`;

exports[`ReposList renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "display": "flex",
        "flex": 1,
        "flexDirection": "column",
        "flexGrow": 1,
        "padding": 40,
      },
    ]
  }
>
  <View>
    <RCTScrollView
      style={
        Object {
          "display": "flex",
          "flex": 1,
          "flexDirection": "column",
        }
      }
    >
      <View>
        <View
          data-testID="repo-215071064"
        >
          <Text
            data-testID="select-repo"
            onPress={[Function]}
            style={
              Array [
                Object {
                  "color": "#333",
                },
                Object {
                  "display": "flex",
                  "flex": 1,
                  "flexWrap": "wrap",
                  "fontSize": 18,
                  "fontWeight": "normal",
                  "lineHeight": 20,
                  "marginTop": 18,
                },
                Object {
                  "color": "#056aff",
                },
              ]
            }
          >
            amadeus-bot
             
          </Text>
        </View>
        <View
          data-testID="repo-155540694"
        >
          <Text
            data-testID="select-repo"
            onPress={[Function]}
            style={
              Array [
                Object {
                  "color": "#333",
                },
                Object {
                  "display": "flex",
                  "flex": 1,
                  "flexWrap": "wrap",
                  "fontSize": 18,
                  "fontWeight": "normal",
                  "lineHeight": 20,
                  "marginTop": 18,
                },
                Object {
                  "color": "#056aff",
                },
              ]
            }
          >
            andor-vis
             
            (5 open issues)
          </Text>
        </View>
      </View>
    </RCTScrollView>
  </View>
</View>
`;
